README.md

```diff
@@ -54,3 +54,80 @@ spark.optimizeSpatialAuto(sourceTable, outputTable, outputLocation)
 // optimize with the user defined block size
 spark.optimizeSpatial(sourceTable, outputTable, outputLocation, blockSize = 20097000)
 ```
+
+
+### Enabling Carto Query Optimizations on Databricks
+
+> This section and approach are based on [docs](https://sedona.apache.org/setup/databricks/)
+> from Apache Sedona.
+
+
+#### Create Initialization Script
+
+First, write a script to DBFS which can be used to copy jars from
+[DBFS](https://docs.databricks.com/data/databricks-file-system.html) to the local filesystem
+on master.
+
+(Note that you will replace 'analytics-toolbox.jar' with whatever your jar is named.)
+This script can be written using a notebook cell:
+```bash
+%sh 
+
+# Create init script directory for Carto
+mkdir -p /dbfs/FileStore/carto/
+
+# Create init script
+cat > /dbfs/FileStore/carto/carto-init.sh <<'EOF'
+#!/bin/bash
+#
+# 
+# On cluster startup, this script will copy the Carto jars to the cluster's default jar directory.
+# In order to activate Carto ST_Intersection plan optimization: "com.carto.analyticstoolbox.spark.rules.SpatialFilterPushdownRules"
+
+# cp /dbfs/FileStore/jars/maven/com/carto/analyticstoolbox/*<version>.jar /databricks/jars
+# cp /dbfs/FileStore/jars/maven/com/carto/analyticstoolbox/*<version>.jar /databricks/jars
+
+# tmp solution to handle the assembly jar
+cp /dbfs/FileStore/jars/maven/com/carto/analyticstoolbox/analytics-toolbox.jar /databricks/jars
+cp /dbfs/FileStore/jars/*<version>.jar /databricks/jars
+
+EOF
+```
+
+
+#### Update Cluster Configuration
+
+Next, we need to update the spark config and inform the cluster of its new initialization script.
+Navigate to cluster settings and find the 'Advanced options'. From your cluster configuration
+activate the Carto Spatial optimizations by adding to the Spark Config
+(`Cluster` -> `Edit` -> `Configuration` -> `Advanced options` -> `Spark`).
+
+To the spark config, add
+```cfg
+spark.sql.extensions com.carto.analyticstoolbox.spark.sql.SpatialFilterPushdownOptimizations
+```
+
+This will inform spark of the class which will register sql extensions. Move from the 'Spark' tab
+of advanced options to the 'Init scripts' tab and add an entry for the initialization script
+written above (dbfs:/FileStore/carto/carto-init.sh).
+
+That should do it. Restart the cluster and predicate pushdown for spatial intersection is enabled,
+allowing certain workflows to run far more efficiently.
+
+
+#### Why is this necessary?
+
+Installing and registering spark optimizations can be tricky. As is often the case when
+working with distributed systems, there are some fundamental sequencing issues that are
+important to understand. Roughly, the startup of a databricks cluster looks something like
```

Comment:
I think that in our case we can be more explicit, this is a Spark specifics; actually any library would behave the same way :shrug:

Reply:
I'm not sure I understand what should be more explicit.  Do you mean changing the list below to the three points you lay out?

Reply:
@moradology Let's remove a part about "...distributed sytems...", in this case it's just how Spark / all JVM apps behave, not related to distributed systems. Or completely remove it or make it less abstract mb 👀 ?

Reply:
Mb just "The startup of a databricks cluster looks something like:" ?

Reply:
It appears you've got a taste for desert landscapes! I agree though: terser is better